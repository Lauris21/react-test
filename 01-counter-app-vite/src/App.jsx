//import FirstComponent from "./FirstComponent";

import  PropTypes  from "prop-types";

const person = {
    name: "Laura",
    age: 30
}

const getPerson = (person) => {
    return JSON.stringify(person)
}
export const App = ({title}) => {
   
  return (
    <>
      <h1>ðŸ’œ{title}</h1>
      <h2>{getPerson(person)}</h2>
      {/* <FirstComponent /> */}
    </>
  );
};

App.propTypes = {
    title: PropTypes.string.isRequired
}

// App.defaultProps = {
    
// }
