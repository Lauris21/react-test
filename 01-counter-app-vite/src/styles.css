body, html {
    color: white;
    background-color: #21232A;
    font-family: Helvetica, Arial, Helvetica, sans-serif;
    font-size: 1.3rem;
    padding: 70px;
}

button {
    padding: 1rem;
}