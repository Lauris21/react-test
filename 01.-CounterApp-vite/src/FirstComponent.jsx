import PropTypes from "prop-types";
import React from "react";

const FirstComponent = ({ title, subtitle, name }) => {
  return (
    <div>
      <h1>FirstComponent</h1>
      <h2 data-testid="test-title">{title}</h2>
      <p>{subtitle}</p>
      <p>{name}</p>
    </div>
  );
};

FirstComponent.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string,
  name: PropTypes.string,
};

export default FirstComponent;
