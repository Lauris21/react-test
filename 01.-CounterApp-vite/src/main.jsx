import React from "react";
import ReactDOM from "react-dom/client";
// import { App } from "./App";
import "./styles.css";
// import { CounterApp } from "./CounterApp";
import FirstComponent from "./FirstComponent";

ReactDOM.createRoot(document.getElementById("root")).render(
  <React.StrictMode>
    {/* <App title={"Hola pipin"}/> */}
    {/* <CounterApp value={3} /> */}

    <FirstComponent title="Hola, Soy tu padre" />
  </React.StrictMode>
);
