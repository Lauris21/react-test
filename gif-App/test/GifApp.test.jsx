import { render, screen } from "@testing-library/react";
import { GifApp } from "../src/GifApp";

describe("Pruebas en <GifApp />", () => {
  test("DeberÃ­a...", () => {
    const { container } = render(<GifApp />);

    expect(container).toMatchSnapshot();
  });
});
