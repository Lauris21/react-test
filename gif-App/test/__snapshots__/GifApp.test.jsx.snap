// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifApp /> Deber√≠a... 1`] = `
<div>
  <h1>
    GifApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="search gif"
      type="text"
      value=""
    />
  </form>
  <h3>
    cats
  </h3>
  <h3>
    Cargando...
  </h3>
  <div
    class="card-grid"
  />
</div>
`;
